pkg/formats/formats.go

```diff
@@ -77,3 +99,35 @@ func (f *Format) Type() string {
 	}
 	return ""
 }
+
+type FormatOpt struct {
+	FormatType    string `yaml:"format-type,omitempty" json:"format-type,omitempty"`
+	FormatVersion string `yaml:"format-version,omitempty" json:"format-version,omitempty"`
+	MimeFormat    string `yaml:"mime-format,omitempty" json:"mime-format,omitempty"`
+}
+
+func (f *FormatOpt) Select() (Format, error) {
```

Comment:
What is the purpose of the Select() function? As it stands it just round trips and returns a format, why not just use the format.Format type directly? The whole formatOpt -> format round trip is just equivalent to just `var f format.Format("....")`. no?

Reply:
I guess its my bad naming of your toFormat